File: evcs-tenant/src/main/java/com/evcs/tenant/entity/Tenant.java
package com.evcs.tenant.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.evcs.common.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 租户实体
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("sys_tenant")
public class Tenant extends BaseEntity {
    
    /**
     * 租户编码
     */
    private String tenantCode;
    
    /**
     * 租户名称
     */
    private String tenantName;
    
    /**
     * 父租户ID（实现分层结构）
     */
    private Long parentId;
    
    /**
     * 祖级列表（逗号分隔的ID）
     */
    private String ancestors;
    
    /**
     * 联系人
     */
    private String contactPerson;
    
    /**
     * 联系电话
     */
    private String contactPhone;
    
    /**
     * 联系邮箱
     */
    private String contactEmail;
    
    /**
     * 企业地址
     */
    private String address;
    
    /**
     * 统一社会信用代码
     */
    private String socialCode;
    
    /**
     * 营业执照URL
     */
    private String licenseUrl;
    
    /**
     * 租户类型：1-平台方，2-运营商，3-第三方合作伙伴
     */
    private Integer tenantType;
    
    /**
     * 状态：0-禁用，1-启用
     */
    private Integer status;
    
    /**
     * 过期时间
     */
    private java.time.LocalDateTime expireTime;
    
    /**
     * 最大用户数
     */
    private Integer maxUsers;
    
    /**
     * 最大充电站数
     */
    private Integer maxStations;
    
    /**
     * 最大充电桩数
     */
    private Integer maxChargers;
    
    /**
     * 备注
     */
    private String remark;
}

File: evcs-admin/src/api/tenant.ts
import request from '../utils/request'
import type { PageResult } from './types'

/**
 * 租户相关接口
 */

// 租户列表查询参数
export interface TenantQueryParams {
  tenantName?: string
  tenantCode?: string
  tenantType?: number
  parentId?: number
  status?: number
  page?: number
  size?: number
}

// 租户信息
export interface Tenant {
  id: number
  tenantCode: string
  tenantName: string
  tenantType: number
  parentId?: number
  parentName?: string
  tenantTypeName?: string
  childrenCount?: number
  contactName?: string
  contactPhone?: string
  contactEmail?: string
  address?: string
  status: number
  maxUsers?: number
  maxStations?: number
  maxChargers?: number
  expireTime?: string
  remark?: string | null
  createTime?: string
  updateTime?: string
  children?: Tenant[]
}

// 租户表单数据
export interface TenantForm {
  tenantCode: string
  tenantName: string
  tenantType: number
  parentId?: number
  contactName?: string
  contactPhone?: string
  contactEmail?: string
  address?: string
  status: number
  maxUsers?: number | null
  maxStations?: number | null
  maxChargers?: number | null
  expireTime?: string | null
  remark?: string | null
}

/**
 * 获取租户列表
 */
export function getTenantList(params: TenantQueryParams) {
  return request<PageResult<Tenant>>({
    url: '/tenant/page',
    method: 'get',
    params
  })
}

/**
 * 获取租户详情
 */
export function getTenantDetail(id: number) {
  return request<Tenant>({
    url: `/tenant/${id}`,
    method: 'get'
  })
}

/**
 * 新增租户
 */
export function createTenant(data: TenantForm) {
  return request<void>({
    url: '/tenant',
    method: 'post',
    data
  })
}

/**
 * 更新租户
 */
export function updateTenant(id: number, data: TenantForm) {
  return request<void>({
    url: `/tenant/${id}`,
    method: 'put',
    data
  })
}

/**
 * 删除租户
 */
export function deleteTenant(id: number) {
  return request<void>({
    url: `/tenant/${id}`,
    method: 'delete'
  })
}

/**
 * 获取租户树形结构
 */
export function getTenantTree() {
  return request<Tenant[]>({
    url: '/tenant/tree',
    method: 'get'
  })
}

/**
 * 获取租户统计信息
 */
export function getTenantStatistics() {
  return request({
    url: '/tenant/statistics',
    method: 'get'
  })
}
